File: Project/PluginDev/lite/base/src/main/java/com/larry/lite/base/LitePlugin.java
package com.larry.lite.base;

import java.io.IOException;

public interface LitePlugin {

    void onCreate(LitePluginPeer litePluginPeer);

    int execute(LitePluginPeer litePluginPeer, LiteConnectionFactory liteConnectionFactory) throws IOException;

    void onDestroy(LitePluginPeer litePluginPeer);

}

File: Project/PluginDev/lite/base/src/main/java/com/larry/lite/base/LitePlugin.kt
package com.larry.lite.base

import java.io.IOException

interface LitePlugin {

    fun onCreate(litePluginPeer: LitePluginPeer)

    @Throws(IOException::class)
    fun execute(litePluginPeer: LitePluginPeer, liteConnectionFactory: LiteConnectionFactory): Int

    fun onDestroy(litePluginPeer: LitePluginPeer)

}
